src/rust/fs/store/src/lib.rs

```diff
@@ -1433,17 +1433,32 @@ impl Store {
                         destination.display()
                     )
                 })?;
-                tokio::fs::set_permissions(&destination, FSPermissions::from_mode(mode))
+                set_permissions(&destination, FSPermissions::from_mode(mode))
                     .await
                     .map_err(|e| {
                         format!(
-                            "Error setting permissions on {}: {e}",
+                            "Error setting permissions on file at {}: {e}",
                             destination.display()
                         )
                     })?;
                 Ok(())
             }
             None => {
+                // If the destination already exists and is readonly, make it writeable so
+                // we can overwrite it.
+                let existing_permissions = match metadata(&destination).await {
```

Comment:
I hate Rust shadowing about half the time. I find the naming convention here a bit confusing.

We pass in `perms` to the function, then we check if they're writable, if not, we apply write to owner/group/world and return that as "existing_perms".

Then after writing - we re-apply the `mode` on top of the `existing_permissions`.

That feels weird to me, but it might just be the naming convention used as `existing_permissions` is more like .... I dunno... `writable_permissions`?

Reply:
I also think "perms" is too wildly used, for not always referencing the same thing. In the loop, would prefer something like `dest_perms` just for explicitness - but these are mostly nits

Reply:
I've renamed to attempt to clarify what's going on. I've also set the writeable mode on the cloned permissions, to emphasize this.